liblocation/src/main/java/com/mapbox/android/core/location/GpxAnimator.java

```diff
@@ -0,0 +1,93 @@
+package com.mapbox.android.core.location;
+
+import android.animation.TimeAnimator;
+import android.location.Location;
+import android.os.Build;
+import android.support.annotation.NonNull;
+import android.support.annotation.RequiresApi;
+import android.util.Log;
+
+import java.io.InputStream;
+import java.util.List;
+import java.util.concurrent.CopyOnWriteArrayList;
+
+/**
+ * This class can be used to replay ".gpx" files, returning {@link Location} objects
+ * with {@link GpxTimeListener}.
+ *
+ * The constructor takes an {@link InputStream} and {@link OnGpxAnimatorReadyCallback}.  Once the
+ * stream has finished being parsed, the callback will be triggered indicating you can begin GPX playback
+ * with {@link GpxAnimator#start()}.
+ *
+ * The {@link InputStream} will be automatically closed once the parsing has finished.
+ *
+ * @since 0.3.0
+ */
+@RequiresApi(api = Build.VERSION_CODES.JELLY_BEAN)
+public class GpxAnimator extends TimeAnimator {
+
+  private static final String TAG = GpxAnimator.class.getSimpleName();
```

Comment:
users in the past have reached out about the `class.getSimpleName()` construct being expensive as it uses reflection